osf/management/commands/sync_doi_metadata.py

```diff
@@ -69,6 +69,47 @@ def sync_doi_metadata(modified_date, batch_size=100, dry_run=True, sync_private=
             sync_identifier_doi.apply_async(kwargs={'identifier_id': identifier.id})
 
 
+@app.task(name='osf.management.commands.sync_doi_empty_metadata_dataarchive_registrations_command', max_retries=5, default_retry_delay=RATE_LIMIT_RETRY_DELAY)
+def sync_doi_empty_metadata_dataarchive_registrations(modified_date, batch_size=100, dry_run=True, sync_private=False, rate_limit=100):
+    registrations_ids = list(
+        Registration.objects.filter(
+            provider___id='dataarchive',
+            is_public=True,
+            deleted__isnull=True,
+        ).values_list('id', flat=True)
+    )
+    identifiers = Identifier.objects.filter(
+        object_id__in=registrations_ids,
+        content_type_id=ContentType.objects.get_for_model(Registration).id,
+        category='doi',
+        deleted__isnull=True,
+        modified__lte=modified_date
+    )
+    if batch_size:
+        identifiers = identifiers[:batch_size]
+        rate_limit = batch_size if batch_size > rate_limit else rate_limit
+
+    logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
+                f'{identifiers.count()} identifiers to mint')
+
+    for record_number, identifier in enumerate(identifiers, 1):
+        if dry_run:
+            logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
+                        f' doi minting for {identifier.value} started')
+            continue
```

Comment:
This should probably be moved to L110, because not every one of the items in this part of the loop will be affected by the management command.